internal/util/util.go

```diff
@@ -326,12 +326,27 @@ func checkDirExists(p string) bool {
 
 // IsMountPoint checks if the given path is mountpoint or not.
 func IsMountPoint(mounter mount.Interface, p string) (bool, error) {
-	notMnt, err := mounter.IsLikelyNotMountPoint(p)
+	// mounter.IsLikelyNotMountPoint(p) is fast but unreliable. Check
+	// if we have a mountpoint, otherwise parse /proc/self/mountinfo
+	// to be sure.
+
+	notMnt, statErr := mounter.IsLikelyNotMountPoint(p)
+	if statErr == nil && !notMnt {
+		return true, nil
+	}
+
+	procMountInfo, err := ReadMountInfoForProc("self")
```

Comment:
Would this work when the mount was done through `nsenter`? It is used for mounting in a different (Linux kernel) network namespace, not sure if the mountinfo is available though `self` in that case.

This is done for all mounts/attaching when configured, one example:
https://github.com/ceph/ceph-csi/blob/devel/internal/cephfs/mounter/fuse.go#L74

Reply:
Just to clarify -- since this has been already fixed in ceph-fuse, let's just wait for the new client instead of solving it here as this may be more difficult.